tap_mailchimp_export/streams.py

```diff
@@ -127,11 +127,14 @@ def transform_event(record, campaign, include_sends):
 
     if 'error' in obj.keys():
         raise Exception(record)
+    elif int(obj.get('status', 0)) >= 400:
+        raise Exception(record)
 
     try:
         (email, events), = json.loads(record).items()
     except ValueError:
         events = []
+        email = ''
```

Comment:
The code path where `email` isn't hit will likely never be hit again. My best guess is that it was simply unhandled because `elif int(obj.get('status', 0)) >= 400:` wasn't there when receiving an error API response without an `error` key in the response body.

However, I'm not sure if this process should throw when `email` is undefined and I'll leave that open for discussion with the reviewer.